base_binary_url_import/wizard/base_binary_url_import.py

```diff
@@ -0,0 +1,302 @@
+            "res_model": self.target_model_id.model,
+            "domain": domain,
+            "view_mode": "tree,form",
+        }
+
+
+class BaseBinaryURLImportLine(models.TransientModel):
+
+    _name = "base.binary.url.import.line"
+    _description = "Import of binaries from URL wizard line"
+
+    wizard_id = fields.Many2one("base.binary.url.import", required=True)
+    target_record_identifier = fields.Char(string="Target record ID")
+    is_target_record_identifier_required = fields.Boolean(
+        compute="_compute_is_target_record_identifier_required"
+    )
+    binary_url_to_import = fields.Char(string="URL to import", required=True)
+
+    @api.depends("binary_url_to_import")
+    def _compute_is_target_record_identifier_required(self):
+        for line in self:
+            line.is_target_record_identifier_required = (
+                SEPARATOR_CHAR not in line.binary_url_to_import
+                if line.binary_url_to_import
+                else False
+            )
+
+    def _get_identifier_type(self):
+        self.ensure_one()
+        target_id = self.target_record_identifier
+        if not target_id:
+            return
+        if target_id.isdigit():
+            return "db_id"
+        elif "." in target_id:
+            return "xml_id"
+        else:
+            raise UserError(_("Identifier %s is not an Integer or XMLID") % target_id)
+
+    def _get_target_record(self):
+        self.ensure_one()
+        rec = None
+        if self._get_identifier_type() == "db_id":
+            rec = self.env[self.wizard_id.target_model_id.model].browse(
+                int(self.target_record_identifier)
+            )
+        elif self._get_identifier_type() == "xml_id":
+            rec = self.env.ref(self.target_record_identifier)
+        return rec
+
+    def import_binary_from_url(self, request_session=None):
+        self.ensure_one()
+        if request_session is None:
+            request_session = requests
+        binary_content, filename = self._import_content_from_url(request_session)
+        target_record = self._get_target_record()
+        target_record.write(
+            {
+                self.wizard_id.target_binary_field_id.name: base64.b64encode(
+                    binary_content
+                ),
+                self.wizard_id.target_binary_filename_field_id.name: filename,
+            }
+        )
+        return target_record.id
+
+    def _import_content_from_url(self, request_session):
+        # Adapted from base_import module
+        maxsize = (
+            int(
+                self.env["ir.config_parameter"].get_param("binary.url.import.max.size")
+                or "0"
+            )
+            or DEFAULT_BINARY_IMPORT_MAXBYTES
+        )
+        timeout = (
+            int(
+                self.env["ir.config_parameter"].get_param("binary.url.import.timeout")
+                or "0"
+            )
+            or DEFAULT_BINARY_IMPORT_TIMEOUT
+        )
```

Comment:
```suggestion
        params = self.env["ir.config_parameter"]
        maxsize = int(
            params.get_param(
                "binary.url.import.max.size",
                DEFAULT_BINARY_IMPORT_MAXBYTES
            )
        )
        timeout = int(
            params.get_param(
                "binary.url.import.timeout",
                DEFAULT_BINARY_IMPORT_TIMEOUT
            )
        )
```

A few refactoring as this is allowed : https://github.com/odoo/odoo/blob/6310ccac71a370ce6cd3011c3d03089d792b7347/odoo/addons/base/models/ir_config_parameter.py#L58